addons/auto-follow-topics/module.js

```diff
@@ -0,0 +1,30 @@
+const getTopics = () => JSON.parse(sessionStorage.getItem("auto-follow-topics") || "[]");
+const setTopics = (topics) => sessionStorage.setItem("auto-follow-topics", JSON.stringify(topics));
+const postError = () => Boolean(document.querySelector(".errorlist"));
+const topicID = location.href.split("/")[5];
+
+document.querySelector("[name=AddPostForm]")?.addEventListener("click", (event) => {
+  const ids = getTopics();
+  ids.push(topicID);
+  setTopics(ids);
+  onPostCallbacks.forEach((callback) => callback());
+});
+
+const topics = getTopics();
+let success = false;
+if (topics.includes(topicID) && !postError()) {
+  success = true;
+  setTopics(topics.filter((topic) => topic !== topicID));
+}
+
+const onPostCallbacks = [];
+export const onPost = (callback) => {
+  onPostCallbacks.push(callback);
+};
+
+export const onSuccessfulPost = (callback) => {
+  if (success) callback(topicID);
+};
+export const onNonSuccessfulPost = (callback) => {
```

Comment:
What does a "non successful post" actually mean? Why are you clearing storage (in forum-post-countdown) in this case? 🤔

Reply:
The variable `success` being false does not mean there was a post error. It only means there was not a successful post. But visiting a different topic shouldn't clear the countdown, right?

Reply:
Thanks, you're right. I fixed that in https://github.com/ScratchAddons/ScratchAddons/pull/7273/commits/4d3a08b6574be21497d62eddc757a6c189f215ba.